main.tf

```diff
@@ -0,0 +1,143 @@
+#####
+# Backup Vault
+#####
+resource "aws_backup_vault" "main" {
+  count = var.vault_name != null ? 1 : 0
+
+  name        = var.vault_name
+  kms_key_arn = var.vault_kms_key_arn
+  tags        = var.tags
+}
+
+#####
+# Backup Plan
+#####
+resource "aws_backup_plan" "main" {
+  name = var.plan_name
+
+  dynamic "rule" {
+    for_each = var.rules
+    content {
+      rule_name           = lookup(rule.value, "name")
+      target_vault_name   = lookup(rule.value, "target_vault_name", null) == null ? var.vault_name : lookup(rule.value, "target_vault_name", "Default")
+      schedule            = lookup(rule.value, "schedule", null)
+      start_window        = lookup(rule.value, "start_window", null)
+      completion_window   = lookup(rule.value, "completion_window", null)
+      recovery_point_tags = length(lookup(rule.value, "recovery_point_tags")) == 0 ? var.tags : lookup(rule.value, "recovery_point_tags")
+
+      dynamic "lifecycle" {
+        for_each = length(lookup(rule.value, "lifecycle")) == 0 ? [] : [lookup(rule.value, "lifecycle", {})]
+        content {
+          cold_storage_after = lookup(lifecycle.value, "cold_storage_after", 0)
+          delete_after       = lookup(lifecycle.value, "delete_after", 90)
+        }
+      }
+
+      dynamic "copy_action" {
+        for_each = length(lookup(rule.value, "copy_action", {})) == 0 ? [] : [lookup(rule.value, "copy_action", {})]
+        content {
+          destination_vault_arn = lookup(copy_action.value, "destination_vault_arn", null)
+
+          dynamic "lifecycle" {
+            for_each = length(lookup(copy_action.value, "lifecycle", {})) == 0 ? [] : [lookup(copy_action.value, "lifecycle", {})]
+            content {
+              cold_storage_after = lookup(lifecycle.value, "copy_action_cold_storage_after", 0)
+              delete_after       = lookup(lifecycle.value, "copy_action_delete_after", 90)
+            }
+          }
+        }
+      }
+    }
+  }
+
+  dynamic "advanced_backup_setting" {
+    for_each = var.advanced_backup_settings
+    content {
+      resource_type = lookup(advanced_backup_setting.value, "resource_type", "EC2")
+      backup_options = {
+        WindowsVSS = lookup(advanced_backup_setting.value, "backup_options", "enabled")
+      }
+    }
+  }
+
+  tags = var.tags
+
+  depends_on = [aws_backup_vault.main]
```

Comment:
`aws_backup_vault` is conditional as you can pass in `var.vault_name`. I don't believe you need that depends_on statement